lib/chef/api_client.rb

```diff
@@ -124,14 +124,21 @@ def to_json(*a)
       Chef::JSONCompat.to_json(to_hash, *a)
     end
 
-    def self.json_create(o)
-      client = Chef::ApiClient.new
-      client.name(o["name"] || o["clientname"])
-      client.private_key(o["private_key"]) if o.key?("private_key")
-      client.public_key(o["public_key"])
-      client.admin(o["admin"])
-      client.validator(o["validator"])
-      client
+    class << self
```

Comment:
i'm actually not wild about this idiom, i like being able to look at the `def self.foo`s to see all the class methods.  visually this makes them stand out less.